csrc/optimization/mark_alias.cpp

```diff
@@ -17,6 +17,13 @@ void MarkAliasPass::runPass(Fusion* fusion) {
   const AliasAnalysisResult alias_analysis = findAliases(fusion);
   for (TensorView* out :
        ir_utils::filterByType<TensorView>(fusion->outputs())) {
+    // Lazy move: we could check compatibility and only give up when
+    // the allocation domain is incompatible with what we prefer for
+    // aliasing.
+    if (out->hasAllocation()) {
+      continue;
```

Comment:
nitpick: Can we add a debug print here explaining why we are ditching the aliasing.

Reply:
Yep!